### PR TITLE
INGM-321 Create functions to connect to comkit

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Add a feature to connect to a servo using a COM-KIT.

Fixes #INGM-321

### Type of change

- Create connect_servo_comkit function
- Create create_comkit_dictionary function

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

### Documentation
- [x] Update docstrings of every function, method or class that change.
- [x] USe [type hints](https://docs.python.org/3/library/typing.html) for every function and argument.
- [x] Build documentation locally to verify changes.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingeniamotion tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

**Review by WoodElv — APPROVED** [review]

File: ingeniamotion/communication.py [hunk 1]
```diff
@@ -225,6 +226,50 @@ def connect_servo_eoe_service_interface_ip(
...
+        dict_path = create_comkit_dictionary(coco_dict_path, moco_dict_path)
+        self.__servo_connect(
+            ip,
+            dict_path,
+            alias,
+            port,
+            connection_timeout,
+            servo_status_listener=servo_status_listener,
+            net_status_listener=net_status_listener,
+        )
```
Comment (GullinBustin) [inline ingeniamotion/communication.py:271]: After connect removes the dictionary file

File: ingeniamotion/comkit.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    Returns:
+        Path to the COMKIT dictionary.
+
+    """
+    coco_dict_name = Path(coco_dict_path).stem
+    moco_dict_name = Path(moco_dict_path).stem
+    if dest_folder is None:
+        dest_folder = tempfile.gettempdir()
+    dest_path = f"{dest_folder}/{coco_dict_name}-{moco_dict_name}{FILE_EXT_DICTIONARY}"
```
Comment (GullinBustin) [inline ingeniamotion/comkit.py:50]: I think generate this filename is not necessary. You can add the filename as an optional parameter and use a temporary file if filename is None, with a random name.

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/communication.py (hunk 1 above)
Comment (MartinALL91) [reply ingeniamotion/communication.py:272]: Done!

File: ingeniamotion/comkit.py (hunk 2 above)
Comment (MartinALL91) [reply ingeniamotion/comkit.py]: Done!

File: ingeniamotion/comkit.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+
+class CORE(Enum):
+    """Core type enum"""
+
+    COMMUNICATION_CORE = "Coco"
+    MOTION_CORE = "Moco"
+
+
+def create_comkit_dictionary(
```
[the remark this answers is not in the document]
Comment (MartinALL91) [reply ingeniamotion/comkit.py:30]: Done!

File: ingeniamotion/communication.py [hunk 4]
```diff
@@ -225,6 +227,50 @@ def connect_servo_eoe_service_interface_ip(
...
+                status, connection and disconnection.
+
+        Raises:
+            FileNotFoundError: If a dict file doesn't exist.
+            ingenialink.exceptions.ILError: If the servo's IP or port is incorrect.
+        """
+        for dict_path in [coco_dict_path, moco_dict_path]:
+            if not path.isfile(dict_path):
+                raise FileNotFoundError(f"{dict_path} file does not exist!")
+        self.__comkit_dict = create_comkit_dictionary(coco_dict_path, moco_dict_path)
```
Comment (GullinBustin) [inline ingeniamotion/communication.py:263]: The comkit dictionary should not be a class property. You can simply remove the file in the 'connect_servo_comkit' function after "__servo_connect" call

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingeniamotion/communication.py (hunk 4 above)
Comment (MartinALL91) [reply ingeniamotion/communication.py]: Done!

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**